src/rdf4cpp/rdf/parser/RDFFileParser.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef RDF4CPP_RDFFILEPARSER_HPP
+#define RDF4CPP_RDFFILEPARSER_HPP
+
+#include <fstream>
+#include "IStreamQuadIterator.hpp"
+
+namespace rdf4cpp::rdf::parser {
+    /**
+     * Similar to rdf4cpp::rdf::parser::IStreamQuadIterator
+     * Parses the file by the given path and tries to extract Quads given in TURTLE format.
+     *
+     * @note the iterator _starts on_ the first Quad
+     * @note An exhausted iterator becomes the end-of-stream iterator.
+     * @warning The value pointed to by an end-of-stream iterator is undefined
+     *
+     * @example
+     * @code
+     * for (const auto& e : rdf4cpp::rdf::parser::RDFFileParser{"somewhere.ttl"}) {
+     *  if (e.has_value())
+     *     std::cout << e.value();
+     *  else
+     *     std::cout << e.error();
+     * }
+     * @endcode
+     */
+    class RDFFileParser {
+        std::string filePath_;
+        ParsingFlags flags_;
+        storage::node::NodeStorage node_storage_;
+
+    public:
+        explicit RDFFileParser(const std::string& filePath, ParsingFlags flags = ParsingFlags::none(),
+                               storage::node::NodeStorage node_storage = storage::node::NodeStorage::default_instance());
+        explicit RDFFileParser(std::string&& filePath, ParsingFlags flags = ParsingFlags::none(),
+                               storage::node::NodeStorage node_storage = storage::node::NodeStorage::default_instance());
+
+        class Iterator {
+            friend class RDFFileParser;
+            friend bool operator==(const RDFFileParser::Iterator& iter, std::default_sentinel_t) noexcept;
+            std::ifstream stream_;
+            IStreamQuadIterator iter_;
+
+            Iterator(std::ifstream&& stream, IStreamQuadIterator&& iter);
+            Iterator(std::ifstream&& stream, ParsingFlags flags, storage::node::NodeStorage node_storage);
+
+        public:
+            using value_type = IStreamQuadIterator::value_type;
+            using reference = IStreamQuadIterator::reference;
+            using pointer = IStreamQuadIterator::pointer;
+            using difference_type = IStreamQuadIterator::difference_type;
+            using iterator_category = IStreamQuadIterator::iterator_category;
+            using istream_type = IStreamQuadIterator::istream_type;
+
+            reference operator*() const noexcept;
+            pointer operator->() const noexcept;
+            Iterator& operator++();
+            [[nodiscard]] bool operator==(const Iterator& other)const noexcept;
+            // != gets generated by compiler
+        };
+
+        [[nodiscard]] Iterator begin();
+        std::default_sentinel_t end();
```

Comment:
I didn't know they added that to C++20. I always used a bool instead. The bool has one advantage that you can check with `if (bool(iter))` weather an iterator is not at the end yet. So, you don't need an end iterator to compare with. 
Having said that, I think we don't need the bool conversion here and then the `std::default_sentinel_t` is the much cleaner standard way of doing it.